zbus/src/connection/builder.rs

```diff
@@ -412,13 +415,13 @@ impl<'a> Builder<'a> {
             for (path, interfaces) in &self.interfaces {
                 for (name, iface) in interfaces {
                     let iface = iface.clone();
-                    let future =
-                        object_server
-                            .inner()
-                            .at_ready(path.to_owned(), name.clone(), || iface);
-                    let added = future.await?;
-                    // Duplicates shouldn't happen.
-                    assert!(added);
+                    let added = object_server
+                        .inner()
+                        .at_ready(path.to_owned(), name.clone(), || iface)
+                        .await?;
+                    if !added {
+                        return Err(Error::Unsupported);
```

Comment:
best add an error variable for this. `Unsupported` seems misplaced since this is a valid failure and not some lack of support of a feature that should be supported.

Reply:
yeah, I wasn't so sure it is not worth to support. But maybe it can be changed anyway another day.

Reply:
We both know that that day will likely never come. :) Let's just add a variant:

```rust
enum Error {
    // ...
    /// The given interface already exists at the given path. 
    InterfaceExists(InterfaceName<'static>, ObjectPath<'static>),
}
```

Not the best name or description, I know but I think it's still better than just throwing a generic error.

Reply:
Our error struct is declared `non_exhaustive` so we should make use of that and added variants wherever needed.